spark/src/main/scala/org/apache/mahout/drivers/ItemSimilarityDriver.scala

```diff
@@ -164,12 +154,13 @@ object ItemSimilarityDriver extends MahoutSparkDriver {
         datasetA.dfsWrite(parser.opts("output").asInstanceOf[String] + "../input-datasets/primary-interactions",
           schema = writeSchema)
 
-      // The case of reading B can be a bit tricky when the exact same row IDs don't exist for A and B
-      // Here we assume there is one row ID space for all interactions. To do this we calculate the
-      // row cardinality only after reading in A and B (or potentially C...) We then adjust the
-      // cardinality so all match, which is required for the math to work.
-      // Note: this may leave blank rows with no representation in any DRM. Blank rows need to
-      // be supported (and are at least on Spark) or the row cardinality adjustment will not work.
+      /** The case of reading B can be a bit tricky when the exact same row IDs don't exist for A and B
```

Comment:
yes, the only comment guidelines are in the scaladoc section of the style guide so I follow them everywhere except a few one liners.